### PR TITLE
Renames Pubsub to PubSub once again

### LINKED ISSUE #80: Refactor Pubsub* to PubSub* (revert #56)
Based on discussions with the Cloud Pub/Sub team, I submitted [an issue on googleapis ](https://github.com/googleapis/googleapis/issues/414) to have the PubSub proto refactored to PubSub instead of Pubsub. That probably isn't going to happen anytime soon due to all the dependencies, given this code was out there for 2 years now. However we can stop the pattern propagating and follow the style guide for the correct camel case.
note: this would be essentially the reversion of #56

### PR DESCRIPTION
**Pull request by joaoandremartins**
Even though #56 uniformized around Pubsub, we concluded that this is
an issue that needs to be solved in the Google Pub/Sub proto. PubSub
is the correct casing of the term, and we should adopt it from early on
here.

Fixes #80

### CODE REVIEW

File: spring-cloud-gcp-pubsub/pom.xml [hunk 1]
```diff
@@ -10,8 +10,8 @@
 	</parent>
 
 	<artifactId>spring-cloud-gcp-pubsub</artifactId>
-	<name>Spring Cloud GCP Pubsub Module</name>
-	<description>Spring Cloud GCP Pubsub Module</description>
+	<name>Spring Cloud GCP PubSub Module</name>
```
Comment (meltsufin) [inline spring-cloud-gcp-pubsub/pom.xml]: I think "Pub/Sub" is more appropriate here and the line below.

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/pom.xml [hunk 2]
```diff
@@ -7,8 +7,8 @@
 		<version>1.0.0.BUILD-SNAPSHOT</version>
 	</parent>
 	<artifactId>spring-cloud-gcp-starter-pubsub</artifactId>
-	<name>Spring Cloud GCP Pubsub Starter</name>
-	<description>Starter for Google Cloud Pubsub messaging</description>
+	<name>Spring Cloud GCP PubSub Starter</name>
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/pom.xml]: "Pub/Sub" and on the line below.

File: spring-integration-gcp/src/test/java/org/springframework/integration/gcp/outbound/PubSubMessageHandlerTests.java [hunk 3]
```diff
@@ -36,20 +36,20 @@
...
+	private PubSubMessageHandler adapter;
 	private Message<?> message;
 
 	@Before
 	public void setUp() {
 		this.message = new GenericMessage<>("testPayload");
 		when(this.pubsubTemplate.send(eq("testTopic"), eq(this.message)))
 				.thenReturn(new SettableListenableFuture<>());
-		this.adapter = new PubsubMessageHandler(this.pubsubTemplate);
+		this.adapter = new PubSubMessageHandler(this.pubsubTemplate);
```
Comment (meltsufin) [inline spring-integration-gcp/src/test/java/org/springframework/integration/gcp/outbound/PubSubMessageHandlerTests.java]: `pubsubTemplate` -> `pubSubTemplate`
.. and the same on some instances below.

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubSubAutoConfiguration.java [hunk 4]
```diff
@@ -94,8 +94,8 @@ public ChannelProvider publisherChannelProvider() {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public PubsubTemplate pubsubTemplate(PublisherFactory publisherFactory) {
-		return new PubsubTemplate(publisherFactory);
+	public PubSubTemplate pubsubTemplate(PublisherFactory publisherFactory) {
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubSubAutoConfiguration.java]: `pubsubTemplate` -> `pubSubTemplate`

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubSubAutoConfiguration.java [hunk 5]
```diff
@@ -120,10 +120,10 @@ public PublisherFactory defaultPublisherFactory(GcpProjectIdProvider projectIdPr
 
 	@Bean
 	@ConditionalOnMissingBean
-	public PubsubAdmin pubsubAdmin(GcpProjectIdProvider projectIdProvider,
+	public PubSubAdmin pubsubAdmin(GcpProjectIdProvider projectIdProvider,
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubSubAutoConfiguration.java]: `pubsubAdmin` -> `pubSubAdmin`

**Comment by joaoandremartins** [comment]
@artembilan did you have the chance to glance over this? Thanks!

**Comment by artembilan** [comment]
No, I didn't. Sorry.
Why do you need my review for renaming again though?
Why don't go without me for this kind of tasks?
Thanks

**Comment by joaoandremartins** [comment]
@artembilan I'm OK both ways. Just wanted to make sure you were aware of this change before merging, so we didn't have the same issue of merging without approvals as before. :)

**Pull request merged**